File: promptodroid/app/src/main/java/xyz/cybersapien/promptodroid/utils/Utilities.java
package xyz.cybersapien.promptodroid.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;

import xyz.cybersapien.promptodroid.R;

/**
 * Created by ogcybersapien on 31/12/17.
 */

public final class Utilities {

    private Utilities() {
        // Empty private constructor as this class must not be able to initialize data
    }

    public static boolean isInternetConnected(Context context) {
        ConnectivityManager cManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo info = cManager != null ? cManager.getActiveNetworkInfo() : null;
        return info != null && info.isConnected();
    }

    public static String getFormattedDate(long date) {
        DateFormat dateFormat = SimpleDateFormat.getDateInstance();
        return dateFormat.format(new Date(date));
    }

    public static String getWordCount(Context context, String detailText) {
        int words = detailText.split(" ").length;
        return context.getString(R.string.words_string, words);
    }
}

File: promptodroid/app/src/main/java/xyz/cybersapien/promptodroid/utils/Utilities.kt
package xyz.cybersapien.promptodroid.utils

import android.content.Context
import android.net.ConnectivityManager
import xyz.cybersapien.promptodroid.R
import java.text.SimpleDateFormat
import java.util.*

/**
 * Created by ogcybersapien on 31/12/17.
 */

fun isInternetConnected(context: Context): Boolean {
    val cManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager?
    val info = cManager?.activeNetworkInfo
    return info != null && info.isConnected
}

fun getFormattedDate(date: Long): String {
    val dateFormat = SimpleDateFormat.getDateInstance()
    return dateFormat.format(Date(date))
}

fun getWordCount(context: Context, detailText: String): String {
    val words = detailText.split(" ").size
    return context.getString(R.string.words_string, words)
}
